cai_bi/models/dataconsumption/financial/timesheet_entry_aggregates/dim_timesheet_entry_aggregates.sql

```diff
@@ -71,297 +75,889 @@ WITH
+            cast(bill_rate_project_entity * qty as number(38,2)) as bill_amount_project_entity,
+            cast(cost_rate_project_entity * qty as number(38,2)) as cost_amount_project_entity,
+            cast(cost_rate_cola_project_entity * qty as number(38,2)) as cost_amount_cola_project_entity,
 
-            --ISSUE: ACTUAL CONVERSION RATE FROM THE PROJECT CURRENCY TO USD. THIS IS MISSING VALUES AND DOESNT ALWAYS HAVE A RATE.
-            cc_usd.fx_rate_mul as fx_rate_to_usd,
-
-            --THE RATE TO USE FOR CONVERTING PROUECT CURRENCY TO USD, IF THERE IS AN ACTUAL VALUE, USE THAT, OTHERWISE USE THE DEFAULT QUARTERS RATE.
-            ifnull(cc_usd.fx_rate_mul, ac.default_fx_rate_to_usd) as converstion_rate_to_usd,
-
-            --ISSUE: THE CONVERSION RATE FROM THE PROJECT CURRENCY TO THE EMPLOYEE CURRENCY, NEED TO LOOK WHY THIS DOESNT ALWAYS HAVE DATA
-            cc_employee.fx_rate_mul as conversion_rate_to_employee
-           
+            -- get raw data in USD currency rates/numbers
+            'USD' as currency_code_employee_usd,
+            cast(bill_rate_employee_usd as number(38,2)) as bill_rate_usd,
+            cast(employee_pay_hourly_rate_usd as number(38,2)) as cost_rate_usd,
+            cast(employee_pay_hourly_rate_cola_usd as number(38,2)) as cost_rate_cola_usd,
+            cast(bill_rate_usd * qty as number(38,2)) as bill_amount_usd,
+            cast(cost_rate_usd * qty as number(38,2)) as cost_amount_usd,
+            cast(cost_rate_cola_usd * qty as number(38,2)) as cost_amount_cola_usd
         from timesheet_entry as te
         left join sage_intacct_employee as intacct_employee on te.employee_id_intacct = intacct_employee.employeeid
         left join sage_intacct_location as intacct_locations on intacct_employee.locationid = intacct_locations.locationid
-        left join currencies_active as ac on currency_project = ac.currency
-        left join currency_conversion as cc_usd on (te.dte_entry = cc_usd.date and currency_project = cc_usd.frm_curr and cc_usd.to_curr = 'USD')      
-        left join currency_conversion as cc_employee on (te.dte_entry = cc_employee.date and te.currency_iso_code = cc_employee.frm_curr and intacct_employee.currency = cc_employee.to_curr)
         left join date_listings_flattened as w on te.dte_entry = w.dte
     ),
-blank_data as (
+    
+    -- ************************************************************************************************************************************************
+    -- BLANK_DATA: default blank data for employee entity grouping so that all data sets have a value even if 0
+    blank_data as (
         select
-            te.key_employee, te.key_entity, te.employee_id,te.date_group_id, te.date_group_type_id, te.date_group_year,
+            te.key_employee, te.key_entity_employee as key_entity, te.employee_id, te.date_group_id, te.date_group_type_id, te.date_group_year,
```

Comment:
Each of these columns should all be on a new line so it's consistent. It is hard to read as one line